source/adapters/level_zero/usm.cpp

```diff
@@ -597,6 +617,40 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMGetMemAllocInfo(
     ZE2UR_CALL(zeMemGetAddressRange, (Context->ZeContext, Ptr, nullptr, &Size));
     return ReturnValue(Size);
   }
+  case UR_USM_ALLOC_INFO_POOL: {
+    auto UMFPool = umfPoolByPtr(Ptr);
+    if (!UMFPool) {
+      return UR_RESULT_ERROR_INVALID_VALUE;
+    }
+
+    std::shared_lock<ur_shared_mutex> ContextLock(Context->Mutex);
+
+    auto SearchMatchingPool =
+        [](std::unordered_map<ur_device_handle_t, umf::pool_unique_handle_t>
+               &PoolMap,
+           umf_memory_pool_handle_t UMFPool) {
+          for (auto &PoolPair : PoolMap) {
+            if (PoolPair.second.get() == UMFPool) {
+              return true;
+            }
+          }
+          return false;
+        };
+
+    for (auto &Pool : Context->UsmPoolHandles) {
+      if (SearchMatchingPool(Pool->DeviceMemPools, UMFPool)) {
+        return ReturnValue(Pool);
+      }
+      if (SearchMatchingPool(Pool->SharedMemPools, UMFPool)) {
+        return ReturnValue(Pool);
+      }
+      if (Pool->HostMemPool.get() == UMFPool) {
+        return ReturnValue(Pool->HostMemPool.get());
```

Comment:
This should also return `ReturnValue(Pool);`

Reply:
thanks @kswiecicki! fixed.